core/src/main/scala/org/apache/spark/util/SparkUncaughtExceptionHandler.scala

```diff
@@ -26,27 +26,34 @@ import org.apache.spark.internal.Logging
  */
 private[spark] object SparkUncaughtExceptionHandler
   extends Thread.UncaughtExceptionHandler with Logging {
+  private[this] var exitOnException = true
```

Comment:
Instead of using a static variable, I prefer to change `SparkUncaughtExceptionHandler` to a class.